### DISCUSSION TITLE
Show live output from hooks

### DISCUSSION THREAD

**Discussion by felipecrs** [helmfile/helmfile#614] [Ideas]
Similar to the `--enable-live-output` for the helm operations, it would be great if this feature could be extended for the hooks as well.

For example, I have a hook that runs a test, and it takes 30 minutes to run. Without live logs, helmfile just keeps waiting for it to finish without any hint of what's happening under the hoods.

**Comment by yxxhero** [comment]
@felipecrs a good idea. PR is welcome.

**Reply by felipecrs** [reply]
@yxxhero I noticed this issue was closed as resolved... should not it be kept open? Unless it was fixed, but according to my tests I don't think it was.

**Comment by felipecrs** [comment]
Here is my workaround to this:

1. I always run `helmfile` as part of a docker container
1. I keep `hooks.showlogs` as `false`
1. My (bash) hooks have this line at the beginning

```bash
#!/bin/bash

exec &> /proc/1/fd/2
```

This way the hooks stdout and stderr are transmitted in real time to my container's stderr.

**Reply by hwo411** [reply]
Thanks @felipecrs, this is a great idea!

Unfortunately, it doesn't work on MacOS (due to lack of `/proc`) and might not work in some non-containerized environments.

However, based on your idea I came up with a portable version of this idea.

```
#!/bin/sh

{ true >&3; } 2> /dev/null && exec >&3 || true
```

And then you can call helmfile with e.g., `3>&1`. If you do not supply descriptor 3, then regular logs will be output given `showlogs` is true.

Obviously, you can use a different descriptor here.

In your environment you can even run `exec 3>&1` on start and have it always output things to stdout.

**Reply by yxxhero** [reply]
@felipecrs PR is welcome.
